### PR TITLE
Fix #7: Validate that DataReader is not both sortable and limited for GridView

### LINKED ISSUE #7: Query limit not working with pager enabled.
The pagerination pageSize overrides the limit set in the query

```
$query = Objects::find()->orderBy('name ASC')->limit(50);
        $dataProvider = new ActiveDataProvider([
                'query' => $query,
                'pagination' => array('pageSize' =>24)
        ]);
```

The above produces

```
find.count({"ns":"geekobjects.objects","limit":-1,"batchSize":0,"skip":-1,"flags":0,"query":{},"fields":[],"started_iterating":false})

```

Which in turns overrides the limit of 50 and returns all of the collections instead of 50

```
$query = Objects::find()->orderBy('name ASC')->limit(50);
        $dataProvider = new ActiveDataProvider([
                'query' => $query,
                'pagination' => false
        ]);
```

the above produces

```
find({"ns":"geekobjects.objects","limit":50,"batchSize":0,"skip":0,"flags":0,"query":{"$query":{},"$orderby":{"name":1}},"fields":[],"started_iterating":false})
```

which limits the query to 50 items but now there isn't a pager since it was set to false.

I don't know if it was intended to do this / or if this is a bug / or if I'm not doing this correctly.

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: src/GridView.php
Comment (vjik) [inline src/GridView.php:1]: `GridView` can render both limited and sorted data at the same time. But `GridView` should disable sorting feature for data reader with limit. And at the same time header of columns should show current sorting from data reader.

I made a mistake, there is no need an exception here, becuase grid view get sorting configuration from data reader.

**Review by vjik — CHANGES_REQUESTED** [review]

File: src/GridView.php
Comment (samdark) [reply src/GridView.php:1]: Like this? https://github.com/yiisoft/yii-dataview/pull/215/commits/db29fae4d2cfd5d78cc907980fc090d9f1af0623

Comment (rossaddison) [reply src/GridView.php:1]: Does https://github.com/yiisoft/yii-dataview/pull/216 have any relevance here?

Comment (vjik) [reply src/GridView.php:1]: @rossaddison No, that's different.

**Review by vjik — CHANGES_REQUESTED** [review]
It's not enough. Sort don't availaible in header, but use sort parameters from URL.

File: src/BaseListView.php [hunk 1]
```diff
@@ -347,6 +347,11 @@ private function prepareDataReaderByParams(
     ): ReadableDataInterface {
         $dataReader = $this->getDataReader();
 
+        if ($dataReader instanceof LimitableDataInterface && $dataReader->getLimit() !== null) {
```
Comment (samdark) [inline src/BaseListView.php:350]: Should we do it for grid only?

Comment (vjik) [reply src/BaseListView.php:350]: I think, it is actual for all `BaseListView` implementations.

**Review by vjik — CHANGES_REQUESTED** [review]
Yet another problem is founded: when limit is pre-defined links in paginator contains sort parameter.

**Comment by samdark** [comment]
@vjik https://github.com/yiisoft/data/pull/201

**Comment by samdark** [comment]
@vjik that PR is now merged. Anything to add in this one?

**Review by vjik — APPROVED** [review]

**Pull request merged**